libslide/src/diagnostics.rs

```diff
@@ -0,0 +1,51 @@
+use crate::common::Span;
+
+/// The kind of a slide diagnostic.
+pub enum DiagnosticKind {
+    /// An error diagnostic. Generally, this diagnostic should be emitted for unrecoverable errors.
+    /// In other cases, a warning or a note may be more applicable.
+    Error,
+    /// A note diagnostic is a generic annotation with no specific connotation like `error`. It can
+    /// be particularly useful as an associated diagnostic, for example in expanding on a primary
+    /// error.
+    Note,
+}
+
+/// A diagnostic for slide source code.
+pub struct Diagnostic {
```

Comment:
Should there be multiple kinds of diagnostics that inherit from the same base?

Reply:
There is no inheritance in Rust, only composition. So a diagnostic is one of a set of kinds.

Reply:
Oh ok +1 then